### ISSUE TITLE
Focus indicator for input fields and select lists

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyone#458]
**Describe the bug**

Quote from @sarahricker:
_keyboard focus on the search box should also have a focus state, but it seems the input box is getting double boarder while the button gets an inset dashed line._

See https://github.com/WordPress/twentytwentyone/issues/444

If I understood correctly, the focus indicator for text input fields like in the comment form text fields and text area, password and search input field is not enough.

![input-focus](https://user-images.githubusercontent.com/7422055/95946618-36a31600-0ded-11eb-83ce-057a9581c10d.gif)

Visually when an input field is in focus, the background color is changed to white, the border is thicker, and there is a cursor marker. 

The background color change itself is minor and does not comply with any contrast requirements, 
it is decorative, and it is not added to select lists where there is already text content.

Technically, the larger "growing border" is an outline, and for some fields this is combined with an offset.

```
input[type="search"]:focus {
    outline-offset: 0;
}
input[type="text"]:focus, input[type="email"]:focus, input[type="url"]:focus, input[type="password"]:focus, input[type="search"]:focus, input[type="number"]:focus, input[type="tel"]:focus, input[type="date"]:focus, input[type="month"]:focus, input[type="week"]:focus, input[type="time"]:focus, input[type="datetime"]:focus, input[type="datetime-local"]:focus, input[type="color"]:focus, .site textarea:focus {
    color: var(--form--color-text);
    outline: 1px solid var(--form--border-color);
    background: var(--global--color-white);
}
```
See:
https://github.com/WordPress/twentytwentyone/blob/trunk/assets/sass/04-elements/forms.scss#L23

**Comment by carolinan** [comment]
Example where the outline is changed from solid to dotted to look more like the default, and an offset of 4px is added:

![comment-form-focus](https://user-images.githubusercontent.com/7422055/95946957-e9737400-0ded-11eb-89c4-b7c8268d05d1.png)

And an example with a dotted outline with no offset:
![comment-form-focus-no-offset](https://user-images.githubusercontent.com/7422055/95947024-0ad46000-0dee-11eb-8336-4dd426c0892c.png)


To me the first version looks OK, but the version without the offset is more difficult for me to see than the existing style with the growing border.

**Comment by carolinan** [comment]
But the search widget input field does not look as good with the 4px offset because of the button position:

![search-widget-form-focus](https://user-images.githubusercontent.com/7422055/95947243-78808c00-0dee-11eb-948d-aa9badb28130.png)

**Comment by carolinan** [comment]
Select list:

![select-focus](https://user-images.githubusercontent.com/7422055/95947410-d7460580-0dee-11eb-9f89-2bfb4299a709.png)

**Comment by melchoyce** [comment]
If we go with browser defaults in #444, this issue will likely fix itself.

**Comment by carolinan** [comment]
The current styles on the fields would still need to be removed.

**Comment by enriquesanchez** [comment]
I personally don't think there is anything wrong with the current focus styles. It's prominent and obvious. I also don't think there is an issue if the focus style of input fields is not the same as the one for buttons. We ideally want to keep them consistent, but it makes sense to have different styles if it makes the focus indication more visible on buttons.

**Comment by carolinan** [comment]
Current focus with black background:

![black](https://user-images.githubusercontent.com/7422055/96209302-8b6e9a00-0f6f-11eb-954e-1c5e7af731b5.gif)

**Comment by carolinan** [comment]
Closed this as we have received feedback from the accessibility team that the current styles are enough.
